### PR TITLE
Add fish shell completion support

### LINKED ISSUE #926: Add autocompletion for fish shell
### Problem
Currently `stripe-cli` does not support shell autocompletion for [fish shell](https://github.com/fish-shell/fish-shell).

### Feature
Please consider adding an autocompletion for fish-shell. Overall UX would improve for fish shell users.

### PR DESCRIPTION
**Pull request by mnorth-stripe**
### Reviewers
r? @
cc @stripe/developer-products

### Summary

Add fish shell completion to the `stripe completion` command, alongside the existing bash and zsh support. Cobra v1.9.1 (already a dependency) provides `GenFishCompletion` and `GenFishCompletionFile`, so no new dependencies are needed.

Closes #926. Supersedes #1282 (adds test coverage and follows fish naming conventions).

**Changes:**

- **`pkg/cmd/completion.go`** — Add `genFish()` following the `genBash`/`genZsh` pattern, fish detection in `detectShell()`, updated descriptions and error messages. Also fixes pre-existing import grouping (`"runtime"` moved to stdlib block).
- **`pkg/cmd/completion_test.go`** — New test file (first tests for `completion.go`): shell detection (8 cases including edge cases), error paths, stdout generation for all three shells, auto-detection integration, and file creation.
- **`.goreleaser/mac.yml`** — Homebrew formula generates and installs `stripe.fish` via `fish_completion.install`.
- **`.gitignore`** — Add `stripe.fish` alongside existing `stripe-completion.bash`/`stripe-completion.zsh`.

### `stripe completion --help`

```diff
-  Generate bash and zsh completion scripts
+  Generate bash, zsh, and fish completion scripts

       --shell string      The shell to generate completion commands for.
-                          Supports "bash" or "zsh"
+                          Supports "bash", "zsh", or "fish"
```

### `stripe completion --shell fish`

```
Detected `fish`, generating fish completion file: stripe.fish

Suggested next steps:
---------------------
1. Move `stripe.fish` to the fish completions directory:
    mkdir -p ~/.config/fish/completions
    mv stripe.fish ~/.config/fish/completions/stripe.fish

Fish automatically loads completions from this directory, so no additional
configuration is needed. Open a new terminal session and completions will
be available.
```

### Test plan

- [x] `go test ./pkg/cmd/ -run "TestDetectShell|TestSelectShell|TestGenFish"` — 15 tests pass
- [x] `make test` — full suite passes
- [x] `make lint` — passes
- [x] Manual test in fish shell via tmux: subcommand completion with descriptions, flag completion with descriptions, prefix-filtered trigger event completion
- [x] Verified in isolated environment (removed existing completions, confirmed no completions before install, installed ours, confirmed completions work)

### CODE REVIEW

File: pkg/fixtures/triggers/payout.paid.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "path": "/v1/payouts",
+      "method": "post",
+      "params": {
+        "amount": 1100,
+        "currency": "usd",
+        "description": "(created by Stripe CLI)"
+      }
+    }
+  ]
+}
```
[the remark this answers is not in the document]
Comment (mnorth-stripe) [reply pkg/fixtures/triggers/payout.paid.json]: Resolved — fixture files were removed from this PR and moved to their own PRs.

File: pkg/fixtures/triggers.go [hunk 2]
```diff
@@ -16,118 +16,123 @@ var triggers embed.FS
...
+	"invoice.payment_failed":                               "triggers/invoice.payment_failed.json",
+	"invoice.payment_succeeded":                            "triggers/invoice.payment_succeeded.json",
+	"invoice.updated":                                      "triggers/invoice.updated.json",
+	"invoice.deleted":                                      "triggers/invoice.deleted.json",
+	"invoice.voided":                                       "triggers/invoice.voided.json",
+	"invoice.sent":                                         "triggers/invoice.sent.json",
+	"invoice.marked_uncollectible":                         "triggers/invoice.marked_uncollectible.json",
+	"invoiceitem.created":                                  "triggers/invoiceitem.created.json",
+	"invoiceitem.deleted":                                  "triggers/invoiceitem.deleted.json",
+	"invoice_payment.paid":                                 "triggers/invoice_payment.paid.json",
```
[the remark this answers is not in the document]
Comment (mnorth-stripe) [reply pkg/fixtures/triggers.go]: Resolved — fixture files were removed from this PR and moved to their own PRs.

File: pkg/cmd/completion.go [hunk 3]
```diff
@@ -24,16 +23,21 @@ func newCompletionCmd() *completionCmd {
...
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return selectShell(cc.shell, cc.writeToStdout)
 		},
 	}
 
-	cc.cmd.Flags().StringVar(&cc.shell, "shell", "", "The shell to generate completion commands for. Supports \"bash\" or \"zsh\"")
+	cc.cmd.Flags().StringVar(&cc.shell, "shell", "", "Shell to generate completions for: bash, zsh, or fish (auto-detected if omitted)")
 	cc.cmd.Flags().BoolVar(&cc.writeToStdout, "write-to-stdout", false, "Print completion script to stdout rather than creating a new file.")
 
+	//nolint:errcheck // RegisterFlagCompletionFunc only fails if the flag doesn't exist
```
Comment (tomer-stripe) [inline pkg/cmd/completion.go]: Why does this need a `//nolint`? Not sure I follow the reasoning.

File: pkg/cmd/completion.go [hunk 4]
```diff
@@ -136,14 +164,33 @@ func genBash(writeToStdout bool) error {
...
+	}
+
+	if autoDetected {
+		fmt.Println("Detected `fish`, generating fish completion file: stripe.fish")
+	} else {
+		fmt.Println("Generating fish completion file: stripe.fish")
+	}
+
+	// true enables completion descriptions (fish displays them inline during tab-complete)
+	err := rootCmd.GenFishCompletionFile("stripe.fish", true)
```
Comment (tomer-stripe) [inline pkg/cmd/completion.go:180]: bit of a nit: the other two generate `stripe-completion.zsh` (or bash). Should we do `stripe-completion.fish`? Not sure if the patterns for fish require a specific file structure though

File: pkg/cmd/completion.go (hunk 3 above)
Comment (mnorth-stripe) [reply pkg/cmd/completion.go]: Good catch — removed the `//nolint:errcheck` and switched to `_ =` assignment instead. The error can only occur if the flag doesn't exist, and we register it 3 lines above, so swallowing it with `_` is clearer than a lint suppression. Fixed in 65299d4.

File: pkg/cmd/completion.go (hunk 4 above)
Comment (mnorth-stripe) [reply pkg/cmd/completion.go:180]: Fish convention is that completion files go into `~/.config/fish/completions/<command>.fish` — the filename must match the command name exactly (i.e., `stripe.fish`). Fish auto-discovers completions by filename, unlike bash/zsh where the filename is arbitrary. The `stripe-completion.{bash,zsh}` pattern works for those shells because they're sourced explicitly. See [fish docs on completions](https://fishshell.com/docs/current/completions.html).
